iot-hub/Samples/device/EdgeDownstreamDevice/Program.cs

```diff
@@ -0,0 +1,131 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Globalization;
+using System.IO;
+using System.Text;
+using System.Threading.Tasks;
+using System.Security.Cryptography.X509Certificates;
+
+using Microsoft.Azure.Devices.Client;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    class Program
+    {
+
+        // 1) Obtain the connection string for your downstream device and to it
+        //    append this string GatewayHostName=<edge device hostname>;
+        // 2) The edge device hostname is the hostname set in the config.yaml of the Edge device
+        //    to which this sample will connect to.
+        //
+        // The resulting string should look like the following
+        //  "HostName=<iothub_host_name>;DeviceId=<device_id>;SharedAccessKey=<device_key>;GatewayHostName=<edge device hostname>"
+        //
+        // Either set the DEVICE_CONNECTION_STRING environment variable with this connection string
+        // or set it in the launchSettings.json.
+        private static string DeviceConnectionString = Environment.GetEnvironmentVariable("DEVICE_CONNECTION_STRING");
+
+        private static int MESSAGE_COUNT = 10;
```

Comment:
We already have a MessageSample and we shouldn't copy/paste code as it's too hart to maintain. 
Reference / use https://github.com/Azure-Samples/azure-iot-samples-csharp/blob/master/iot-hub/Samples/device/MessageSample/MessageSample.cs